parser_with_temp.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-8 -*-
+"""
+General description:
+The module is purposed for grabbing certain data from tables on different pages
+in the admin part of aimed site.
+First page with general table can be requested with route:"route=sale/order"
+There are two types of tables needed to be parsed:
+1. General tables wich contain general data of an orders. Only order id's to
+be grabbed from these tables. Route for these pages "route=sale/order",
+"&page=" and number of the page.
+2.Full tables page wich contain detailed data of an order can be reached with
+route "index.php?route=sale/order", "&order_id=" and numerical order id grabbed
+from the general tables. All necessary data is received from full tables.
+
+Module iterates a list with numbers of general tables pages,
+takes all order id's than iterates a list with order id's calling pages with
+full tables and grabbing data from the tables into one full data dictionary.
+
+When dictionary is completed module creates an .xlsx file where inserts data
+from the dictionary in certain order.
+"""
+
+from bs4 import BeautifulSoup
+from datetime import datetime
+from os import remove
+from time import time
+from urlparse import urlparse, parse_qs
+import ast
+import re
+import requests
+import sys
+import xlsxwriter
+
+
+class Session(requests.Session):
+    """Class inherited from requests.Session object and has additional methods
+    """
+
+    def login(self, site_url, authentication_data):
+        """Sends a request for getting response with a token from the server
+        creates session properties containing base url and url with token for
+        future usage"""
+        try:
+            self.base_url = site_url
+            self.logined_url = self.post(site_url, authentication_data).url
+        except requests.adapters.ConnectionError:
+            sys.exit(
+                'Connection issues. Check connection and try again')
+
+
+    def get_token(self):
+        """parses token from the response and creates session token property"""
+        try:
+            self.token = '&token=' + parse_qs(urlparse(
+                self.logined_url).query).get('token')[0]
+        except TypeError:
+            sys.exit(
+                'Login issues. Please enter properly username and password.')
+
+    def get_top_number_of_general_page(self):
+        """ sends request for the first page containing general table and grabs
+        the top number of pages containing general tables. Applies
+        top_page_number to the session"""
+        route = 'index.php?route=sale/order'
```

Comment:
The variables "route" in different methods has different values and are used in different scopes. So, I leave it, as local variable.